src/client/providers/execInTerminalProvider.ts

```diff
@@ -38,8 +40,13 @@ function execInTerminal(fileUri?: vscode.Uri) {
         filePath = `"${filePath}"`;
     }
     const terminal = vscode.window.createTerminal(`Python`);
+    if (pythonSettings.terminal.executeInFileDir) {
```

Comment:
@gandhis1 , this line could fall over if such a setting doesn't exist.
You might want to check if it eixsts as follows (or similar way):
```typescript
if (pythonSettings.termal && pythonSettings.terminal.executeInFileDir)
```